### PR TITLE
Modification of the renderer_hasmany

### PR DESCRIPTION
**Pull request by smeeckaert**
I've changed a few things / corrected some bugs with the renderer HasMany
- I check if the values are given before trying to access it
- I target properly in css the header of the hasmany item in order to not affect children
- I allow the overriding of the hasmany content with a view in a renderer option. I also added a view based on divs rather than a table. (can be useful when nesting renderer_hasmany)

### CODE REVIEW

**Comment by jguyomard** [comment]
:+1:

File: static/css/hasmany.css [hunk 1]
```diff
@@ -87,3 +87,19 @@
...
+    width       : 10%;
+    float       : left;
+    font-weight : bold;
+    line-height : 28px;
+}
+
+.hasmany_content_div > .hasmany_content_div_field input, .hasmany_content_div > .hasmany_content_div_field select, .hasmany_content_div  > .hasmany_content_div_field textarea {
+    max-width       : 85%;
+    margin-left : 5%;
+}
```
Comment (shaoshiva) [inline static/css/hasmany.css]: "No newline at end of file" !

File: views/hasmany/content.view.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<table>
+    <?php
+    foreach ($fields as $field) {
+        echo $field->build();
+    }
+    ?>
+</table>
```
Comment (shaoshiva) [inline views/hasmany/content.view.php]: "No newline at end of file" !

File: views/hasmany/content_div.view.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    <?php
+    foreach ($fields as $field) {
+        $tpl = $field->template;
+        if (empty($tpl)) {
+            $field->set_template('<div class="hasmany_content_div_field">{label} {field}</div>');
+        }
+        echo $field->build();
+    }
+    ?>
+</div>
```
Comment (shaoshiva) [inline views/hasmany/content_div.view.php]: "No newline at end of file" !

Comment (smeeckaert) [inline views/hasmany/content_div.view.php]: :fu:

Comment (shaoshiva) [inline views/hasmany/content_div.view.php]: :trollface:

**Comment by shaoshiva** [comment]
:+1:

**Pull request merged**